webapp/src/utils/validation/string.ts

```diff
@@ -183,3 +207,78 @@ function generatePattern(
     allowSpecialChars && specialChars ? escapeSpecialChars(specialChars) : "";
   return basePattern + spacePattern + specialCharsPattern + "]*$";
 }
+
+interface PathValidationOptions {
+  allowToStartWithSlash?: boolean;
+  allowToEndWithSlash?: boolean;
+  allowEmpty?: boolean;
+}
+
+/**
+ * Validates a path against specified criteria.
+ *
+ * @example
+ * validatePath("foo/bar", { allowToEndWithSlash: false }); // true
+ * validatePath("foo/bar/", { allowToEndWithSlash: false }); // Error message
+ *
+ * @example <caption>With currying.</caption>
+ * const fn = validateString({ allowToEndWithSlash: false });
+ * fn("foo/bar"); // true
+ * fn("foo/bar/"); // Error message
+ *
+ * @param path - The string to validate.
+ * @param options - Configuration options for validation. (Optional)
+ * @param [options.allowToStartWithSlash=true] - Indicates if the path is allowed to start with a '/'.
+ * @param [options.allowToEndWithSlash=true] - Indicates if the path is allowed to end with a '/'.
+ * @param [options.allowEmpty=false] - Indicates if an empty path is allowed.
+ * @returns True if validation is successful, or a localized error message if it fails.
+ */
+export function validatePath(
+  path: string,
+  options?: PathValidationOptions,
+): ValidationReturn;
+
+export function validatePath(
+  options?: PathValidationOptions,
+): (value: string) => ValidationReturn;
+
+export function validatePath(
+  pathOrOpts?: string | PathValidationOptions,
+  options: PathValidationOptions = {},
+): ValidationReturn | ((value: string) => ValidationReturn) {
+  if (typeof pathOrOpts !== "string") {
+    return (v: string) => validatePath(v, pathOrOpts);
+  }
+
+  const path = pathOrOpts;
+
+  const {
+    allowToStartWithSlash = true,
+    allowToEndWithSlash = true,
+    allowEmpty = false,
+  } = options;
+
+  if (!path) {
+    return allowEmpty ? true : t("form.field.required");
+  }
+
+  if (!allowToStartWithSlash && path.startsWith("/")) {
+    return t("form.field.path.startWithSlashNotAllowed");
+  }
+
+  if (!allowToEndWithSlash && path.endsWith("/")) {
+    return t("form.field.path.endWithSlashNotAllowed");
+  }
+
+  if (
+    path
+      .replace(/^\/|\/$/g, "") // Remove first "/" and last "/" if present
```

Comment:
What are the fix so? I have no idea of this issue.

Reply:
To fix it you can use native string methods (`startsWith`, `endsWith`, `split`) which are more secure but this will make the code very long to achieve the same thing.. Don't seem a major security issue to me i don't know why Sonar raises errors for this 
maybe you can try this:
```typescript
path = path.startsWith('/') ? path.slice(1) : path;
path = path.endsWith('/') ? path.slice(0, -1) : path;
```
or 
```typescript
path = path.replace(/^\//, "").replace(/\/$/, "");
```

seems like the OR (|) operator is problematic